File: src/main/java/com/zippeykeys/praisebe/block/tile/base/TileEntityIdolBase.java
package com.zippeykeys.praisebe.block.tile.base;

import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.util.ITickable;


public class TileEntityIdolBase extends PBTileEntity implements ITickable{
    private int facing;

    @Override
    public void readFromNBT(NBTTagCompound tagCompound){
        super.readFromNBT(tagCompound);
        facing = tagCompound.getInteger("facing");
    }

    @Override
    public NBTTagCompound writeToNBT(NBTTagCompound tagCompound){
        super.writeToNBT(tagCompound);
        tagCompound.setInteger("facing", facing);
        return tagCompound;
    }

    public int getFacing(){
        return facing;
    }

    public void setFacing(int face){
        facing = face;
    }

    @Override
    public void update(){
    }
}

File: src/main/kotlin/com/zippeykeys/praisebe/block/tile/base/TileEntityIdolBase.kt
package com.zippeykeys.praisebe.block.tile.base

import net.minecraft.nbt.NBTTagCompound
import net.minecraft.util.ITickable


class TileEntityIdolBase : PBTileEntity(), ITickable {
    var facing: Int = 0

    override fun readFromNBT(tagCompound: NBTTagCompound) {
        super.readFromNBT(tagCompound)
        facing = tagCompound.getInteger("facing")
    }

    override fun writeToNBT(tagCompound: NBTTagCompound): NBTTagCompound {
        super.writeToNBT(tagCompound)
        tagCompound.setInteger("facing", facing)
        return tagCompound
    }

    override fun update() {}
}